src/strbuf.c

```diff
@@ -0,0 +1,483 @@
+#include <nitro.h>
+#include <string.h>
+
+#include "charcode.h"
+#include "heap.h"
+#include "strbuf.h"
+
+struct Strbuf_t {
+    u16 maxsize;
+    u16 size;
+    u32 magic;
+    Charcode data[1];
+};
+
+#define SIZEOF_STRBUF_HEADER    (sizeof(Strbuf) - sizeof(Charcode))
+#define STRBUF_MAGIC            (0xB6F8D2EC)
+
+static inline void Strbuf_Check(const Strbuf *strbuf)
+{
+    GF_ASSERT(strbuf != NULL);
+    GF_ASSERT(strbuf->magic == STRBUF_MAGIC);
```

Comment:
Don't follow naming scheme from other repos if the name can be better, because the other repos are volatile and subject to change anyway, plus all three repos currently have vastly different struct names/whatever anyway (so there will need to be one big fix later down the line).